server/src/main/java/org/opensearch/index/IndexModule.java

```diff
@@ -157,7 +157,10 @@ public final class IndexModule {
     /** Which lucene file extensions to load with the mmap directory when using hybridfs store.
      *  This is an expert setting.
      *  @see <a href="https://lucene.apache.org/core/9_2_0/core/org/apache/lucene/codecs/lucene92/package-summary.html#file-names">Lucene File Extensions</a>.
+     *
+     * @deprecated This setting will be removed in 4.x
```

Comment:
Can you also add note on why this setting is deprecated? Like use nio.extensions. And if both are set, nio.extensions will take precedence over mmap.extensions

Reply:
I updated comments for both mmap and nio